beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInvoicePlugin.java

```diff
@@ -0,0 +1,265 @@
+        subscriptionChecker.checkSubscriptionCreated(bpSubscription.getId(), internalCallContext);
+        // Invoice failed to generate
+        assertEquals(invoiceUserApi.getInvoicesByAccount(account.getId(), false, callContext).size(), 0);
+
+        // Verify bus event has moved to the retry service (can't easily check the timestamp unfortunately)
+        checkRetryBusEvents();
+
+        // Add 5'
+        clock.addDeltaFromReality(5 * 60 * 1000);
+        checkRetryBusEvents();
+
+        // Fix invoice plugin
+        testInvoicePluginApi.shouldThrowException = false;
+
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT, NextEvent.PAYMENT);
+        clock.addDeltaFromReality(10 * 60 * 1000);
+        assertListenerStatus();
+        // No notification in the main queue at this point (the PHASE event is the trigger for the next one)
+        checkNotificationsNoRetry(0);
+
+        assertEquals(invoiceUserApi.getInvoicesByAccount(account.getId(), false, callContext).size(), 1);
+        invoiceChecker.checkInvoice(account.getId(),
+                                    1,
+                                    callContext,
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2012, 4, 1), null, InvoiceItemType.FIXED, BigDecimal.ZERO),
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2012, 4, 1), null, InvoiceItemType.TAX, new BigDecimal("1.0")));
+
+        busHandler.pushExpectedEvents(NextEvent.PHASE, NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT, NextEvent.PAYMENT);
+        clock.setDay(new LocalDate("2012-05-01"));
+        assertListenerStatus();
+        checkNotificationsNoRetry(1);
+
+        assertEquals(invoiceUserApi.getInvoicesByAccount(account.getId(), false, callContext).size(), 2);
+        invoiceChecker.checkInvoice(account.getId(),
+                                    2,
+                                    callContext,
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2012, 5, 1), new LocalDate(2012, 6, 1), InvoiceItemType.RECURRING, new BigDecimal("29.95")),
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2012, 5, 1), null, InvoiceItemType.TAX, new BigDecimal("1.0")));
+
+        // Make invoice plugin fail again
+        testInvoicePluginApi.shouldThrowException = true;
+
+        clock.addMonths(1);
+        assertListenerStatus();
+
+        // Invoice failed to generate
+        assertEquals(invoiceUserApi.getInvoicesByAccount(account.getId(), false, callContext).size(), 2);
+
+        // Verify notification has moved to the retry service
+        checkRetryNotifications("2012-06-01T00:05:00");
+
+        // Add 5'
+        clock.addDeltaFromReality(5 * 60 * 1000);
+        checkRetryNotifications("2012-06-01T00:15:00");
```

Comment:
Yes: the delay is added to the initial notification effective date, which is aligned with the reference time of the account in this case (created right after we set the date to `new LocalDate(2012, 4, 1)`).

For bus events, since the initial effective date is computed as `clock.getUTCNow()`, we cannot do this though.